### DISCUSSION TITLE
en/stable/user-guides/script-basics/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj-labs/hera#1020] [Website Discussions]
# en/stable/user-guides/script-basics/

https://hera.readthedocs.io/en/stable/user-guides/script-basics/

**Comment by Yanis-Kouidri** [comment]
I well understand the `InlineScriptConstructor` but I have trouble to understand what `RunnerScriptConstructor` docker image should be. 

Do I need to build a Docker image with python, required modules and the hera script directly on the image or it should be only the core of the function ?

**Reply by elliotgunton** [reply]
>Do I need to build a Docker image with python, required modules and the hera script directly on the image

Yes, this one! 😄 It's designed and intended for you to build the image in your CI pipeline automatically

**Comment by jiantang99** [comment]
Is there a way to create and run a workflow without SSO set up for argo workflow? I left the token as an empty string (i.e., token = ""), and this is the error I get when attempting to run it:
 Error: workflows must use workflowTemplateRef to be executed when the controller is in reference mode

**Reply by jiantang99** [reply]
Edit: the server configured as --auth=server

**Reply by elliotgunton** [reply]
Hey @jiantang99, we do this in our CI checks to run a workflow on an actual cluster. See the setup in https://github.com/argoproj-labs/hera/blob/08038f0a193b8c8eb600f33583377e5ef85019a0/Makefile#L157-L164 and the Python call (note `verify_ssl=False` which may be your issue?) in https://github.com/argoproj-labs/hera/blob/08038f0a193b8c8eb600f33583377e5ef85019a0/tests/test_submission.py#L16-L34
